### PR TITLE
Add command `eksctl utils aws-auth`

### PR DESCRIPTION
**Pull request by rndstr**
Adds the command
```
eksctl utils aws-auth --name=<name> [--add-admin-role=<arn>] [--remove-role=<arn>] [--add-account=<number>] [--remove-account=<number>]
```
to allow modifying the `kube-system/aws-auth` object which is a mapping from IAM entities to Kubernetes groups.

### CODE REVIEW

**Comment by rndstr** [comment]
@mumoshu thanks for the suggestions. this PR is motivated by the former (sorry for scarce PR description). I briefly went over this with @errordeveloper beforehand and we were talking about a separate command.

What I'm personally looking for is the functions to simplify adding/removing roles/accounts, so the command is secondary to me.

I prefer your suggestion over mine (dislike the verbs in my args), so I'll change this.

File: pkg/authconfigmap/authconfigmap.go [hunk 1]
```diff
@@ -1,185 +1,241 @@
...
+	"sort"
 
 	"github.com/kris-nova/logger"
 	"github.com/pkg/errors"
-
-	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha4"
-
-	yaml "gopkg.in/yaml.v2"
-
+	"gopkg.in/yaml.v2"
```
Comment (errordeveloper) [inline pkg/authconfigmap/authconfigmap.go]: While at it, could you please change this to `sigs.k8s.io/yaml`, it's not the preferred version. I believe the interface is 100% the same.

File: pkg/authconfigmap/authconfigmap.go [hunk 2]
```diff
@@ -1,185 +1,241 @@
...
-	mapRolesBytes, err := yaml.Marshal(*mapRoles)
+// AddAccount appends an IAM account to the `mapAccounts` entry
+// in the Configmap. It also deduplicates.
+func (a *AuthConfigMap) AddAccount(account string) error {
+	accounts, err := a.accounts()
 	if err != nil {
-		return nil, err
+		return err
+	}
+	distinct := map[string]struct{}{account: {}}
```
Comment (errordeveloper) [inline pkg/authconfigmap/authconfigmap.go]: Perhaps could use `k8s.io/apimachinery/pkg/util/sets` here?

**Comment by rndstr** [comment]
@errordeveloper what about
```
eksctl <create|get|delete> auth-account
eksctl <create|get|delete> auth-role
```
which elaborates on what kind of account or role this is. 

i first preferred
```
eksctl auth <add-account|remove-account> AWS_ACCOUNT_ID
eksctl auth <add-role|remove-role> ROLE_ARN --username USERNAME --groups GROUP1,GROUP2
```

but I think a command to display the current version would be nice too, which would be `eksctl auth get` which screws over the `eksctl <verb> <subject>` consistency.

**Comment by errordeveloper** [comment]
On Tue, 12 Mar 2019, 9:20 pm Roli Schilter, <notifications@github.com>
wrote:

> @errordeveloper <https://github.com/errordeveloper> what about
>
> eksctl <create|get|delete> auth-account
> eksctl <create|get|delete> auth-role
>
> which elaborates on what kind of account or role this is.
>

Yes, it is odd cause it breaks out a narrow concept of what is a single
object inside the cluster into several new concepts. Most resources we have
at the moment are the opposite, e.g. a cluster brings together great many
things, and so does a nodegroup (albeit to a lesser degree).

Perhaps when we add IAM as a resource (along with VPC), this could be
somehow part of that, or perhaps it will become clearer when we have
'eksclt create|get|delete cluster-iam|nodegroup-iam' (or whatever we decide
to call it).

I would rather merge this PR as is and continue design discussion elsewhere
:)

**Comment by errordeveloper** [comment]
So my conclusion is that we either move along, merger it and review later
on, or move current implementation to 'eksctl update iam-auth-bindings'
(considering that other '*iam*' resources will materialise in the future).
Feel free to keep the flags as they are. If you feel like that would be the
right thing, 'eksclt get iam-auth-bindings' would be nice to have, of
course.

**Comment by rndstr** [comment]
@errordeveloper created https://github.com/weaveworks/eksctl/issues/625 and removed the command itself for now

**Comment by errordeveloper** [comment]
hm... integration tests failed for me, seems like it's to do with this change?
```
2019-03-13T18:48:40Z [▶]  done after 4m22.7808778s of waiting for CloudFormation stack "eksctl-exciting-mushroom-1552501999-nodegroup-ng-0" to reach "CREATE_COMPLETE" status
2019-03-13T18:48:41Z [▶]  processing stack outputs
2019-03-13T18:48:41Z [▶]  task 0 returned without errors
2019-03-13T18:48:41Z [✔]  all EKS cluster resource for "exciting-mushroom-1552501999" had been created
2019-03-13T18:48:41Z [▶]  merging kubeconfig files
2019-03-13T18:48:41Z [▶]  setting current-context to ilya@exciting-mushroom-1552501999.us-west-2.eksctl.io
2019-03-13T18:48:41Z [✔]  saved kubeconfig as "/tmp/kubeconfig"
2019-03-13T18:48:42Z [▶]  aws-auth = {
  TypeMeta: {
    Kind: "",
    APIVersion: ""
  },
  ObjectMeta: {
    Name: "",
    GenerateName: "",
    Namespace: "",
    SelfLink: "",
    UID: ,
    ResourceVersion: "",
    Generation: 0,
    CreationTimestamp: {
      Time: 0001-01-01 00:00:00 +0000 UTC
    },
    ClusterName: ""
  }
}
2019-03-13T18:48:42Z [ℹ]  adding role "arn:aws:iam::376248598259:role/eksctl-exciting-mushroom-15525019-NodeInstanceRole-5EUF1ZLNQ3SG" to auth ConfigMap
2019-03-13T18:48:42Z [✖]  saving auth ConfigMap: ConfigMap "" is invalid: metadata.name: Required value: name or generateName is required
```

@rndstr wdyt?

**Pull request merged**